File: upload-custom-script.java
//@auth
//@required('url', 'scriptName', 'scriptType')

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Collection;
import com.hivext.api.server.development.response.ApplicationInfoResponse;
import com.hivext.api.utils.Random;


//reading script from URL
int READ_TIMEOUT = 30 * 1000;
int CONNECT_TIMEOUT = 10 * 1000;
int BUFFER_SIZE = 2048;

String url = getParam("url");

HttpURLConnection conn = null;
InputStream cis = null;
ByteArrayOutputStream out = null;
try {
    conn = (HttpURLConnection) new URL(url).openConnection();
    conn.setInstanceFollowRedirects(false);
    conn.setReadTimeout(CONNECT_TIMEOUT);
    conn.setConnectTimeout(READ_TIMEOUT);

    byte[] buff = new byte[BUFFER_SIZE];
    cis = conn.getInputStream();
    out = new ByteArrayOutputStream();
    int n;
    while ((n = cis.read(buff)) > -1) {
        out.write(buff, 0, n);
    }
} finally {
    if (cis != null) {
        try {
            cis.close();
        } catch (IOException ex) {}
    }
    if (out != null) {
        try {
            out.close();
        } catch (IOException ex) {}
    }

    if (conn != null) {
        conn.disconnect();
    }

}

//inject token
String token = Random.getPswd(64);
String scriptBody = out.toString("UTF-8").replace("${TOKEN}", token);

//creating a new script 
return hivext.dev.scripting.CreateScript(getParam("scriptName"), getParam("scriptType"), scriptBody);

File: upload-custom-script.cs
//@auth
//@required('url', 'next')

import com.hivext.api.core.utils.Transport;
import com.hivext.api.utils.Random;

//reading script from URL
var scriptBody = new Transport().get(url)

//inject token
var token = Random.getPswd(64);
scriptBody = scriptBody.replace("${TOKEN}", token);

//delete the script if it exists already
jelastic.dev.scripting.DeleteScript(scriptName);

//create a new script 
var envName = "${env.envName}",
    scriptName = envName + "-git-push-redeploy"; 
  
var resp = jelastic.dev.scripting.CreateScript(scriptName, 'js', scriptBody);
if (resp.result != 0) return resp;

//get app domain
var domain = jelastic.dev.apps.GetApp(appid).hosting.domain;

return {
    result: 0,
    onAfterReturn : {
        call : {
            procedure : next,
            params : {
                domain : domain,
                token : token
            }
        }
    }
}
